crates/pairwise_nostd_internal/src/bins.rs

```diff
@@ -0,0 +1,194 @@
+//! Implements types to represent "bins", used for Histogram buckets and for
+//! distance binning of accumulators.  The Bins trait provides a common
+//! interface which is implemented by RegularBins and IrregularBins. These
+//! types do NOT "hold" data, they only define the bin edges and determine the
+//! bin index for a given value.
+
+/// Super simple. This can be expanded as needed.
+pub trait BinEdges {
+    /// Calculate the bin index for a given value. Values which are equal to
+    /// boundary values are considered part of the higher bin.
```

Comment:
To borrow the language from [`numpy.digitize`] is this equivalent to saying that "intervals do not include the right edge?"